### DISCUSSION TITLE
How do you install fast when developing?

### DISCUSSION THREAD

**Discussion by arn4** [ml-explore/mlx#700] [General]
I'm trying to implement new features for mlx, but I'm not to used in developing packages and I find super annoying the fact I have to reinstall the package every little change I do in the code.

More specifically, I'm working on the `py11bind` files, and every edit I do I run `env CMAKE_BUILD_PARALLEL_LEVEL="" env DEBUG="1" pip install -e .` for recompile the package. But this command complies all the files, even the one I did not change and it takes quite a while. How can I reinstall compiling only the changes?

Sorry if it is a very dumb question, but I'm new to Cmake :(

**Comment by awni** [comment]
That's annoying... I didn't realize doing `pip install -e .` would rebuild everything. This one should work:

```
env CMAKE_BUILD_PARALLEL_LEVEL="" python setup.py develop
```

though `setup.py develop` is "deprecated". There might be a workaround using `pip` directly. I'll see if I can find something.

**Reply by arn4** [reply]
Thaks!

Your suggestions works 👍🏻
